Update the `Calendar` API to use a `multiselect` prop instead of `selectionVariant="multiselect"` (#5309)
jpmorganchase/salt-ds: site/docs/components/calendar/examples.mdx

<LivePreview componentName="calendar" exampleName="Single" />

To enable multiple selection of single dates, provide the `multiselect` prop.

<LivePreview componentName="calendar" exampleName="SingleMultiselect" />

To de-select the previous selection, you can use `select` callback to provide control of the current selected dates.

<LivePreview
  componentName="calendar"
  exampleName="SingleMultiselectControlled"
/>

## Date range selection

When the `selectionVariant` prop is set to "range", the `Calendar` component allows users to select a date range, including both a start date and an end date, enabling a comprehensive date selection experience.

> _Multiple selection of date ranges, is not yet supported for J.P. Morgan use-cases, refer to the [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_

Reviewer: ```suggestion
_Multiple selection of date ranges, is not yet supported refer to the [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_
```

> _Multiple selection of date ranges, is not yet supported for J.P. Morgan use-cases, refer to the [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_

Reviewer: Not sure what this means? Is our guidance that we don't recommend its usage?

<LivePreview componentName="calendar" exampleName="Single" />

> To enable multiple selection of single dates, provide the `multiselect` prop.

Reviewer 2: Nit: This slightly differs from the language we typically use in our documentation.

```suggestion
To enable multiple selection of single dates, set the `multiselect` prop to `true`.
```

OR

```suggestion
To enable multiple selection of single dates, you can use the `multiselect` prop.
```

<LivePreview componentName="calendar" exampleName="SingleMultiselect" />

To de-select the previous selection, you can use `select` callback to provide control of the current selected dates.

<LivePreview
  componentName="calendar"
  exampleName="SingleMultiselectControlled"
/>

## Date range selection

When the `selectionVariant` prop is set to "range", the `Calendar` component allows users to select a date range, including both a start date and an end date, enabling a comprehensive date selection experience.

<LivePreview componentName="calendar" exampleName="Range" />

_Multiple selection of date ranges, is still work-in-progress, refer to our [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_

## Offset selection

When the `selectionVariant` prop is set to "offset", the `Calendar` component allows users to select a date range by choosing a single start date, with the end date automatically determined based on a predefined offset.

<LivePreview componentName="calendar" exampleName="Offset" />

> _Multiple selection of date ranges, is still work-in-progress, refer to our [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_

Reviewer 2: ```suggestion
_Multiple selection of date ranges is still work-in-progress, refer to our [roadmap](https://github.com/jpmorganchase/salt-ds/issues/4081) for details_
```
